### PR TITLE
normalize

### PR DESCRIPTION
**Pull request by orthagonal**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -50,6 +50,24 @@ const response = function(fn, options = {}) {
     let res = null;
     const start = new Date().getTime();
     let statusCode = null;
+    // make sure query and path params always exist:
+    req.queryStringParameters = req.queryStringParameters || {};
+    req.pathParameters = req.pathParameters || {};
+    // preserve the original headers
+    req.headersRaw = Object.assign({}, req.headers);
+    // and normalize the headers so you don't have to worry about case:
+    req.headers = Object.keys(req.headersRaw).reduce((memo, key) => {
```
Comment (jgallen23) [inline index.js:59]: any concern lowering them all?  the example had some that it skips

File: index.js [hunk 2]
```diff
@@ -50,6 +50,24 @@ const response = function(fn, options = {}) {
...
+      memo[key.toLowerCase()] = req.headersRaw[key];
+      return memo;
+    }, {});
+    if (req.headers['content-type'] === 'application/json') {
+      try {
+        req.bodyRaw = req.body;
```
Comment (jgallen23) [inline index.js:65]: you need a Buffer.from(req.body).toString() (I believe)

**Review by jgallen23 — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (orthagonal) [reply index.js:59]: They are preserving some of the caps ("WebSocket") etc.  We just want everything to be lowercase characters.

Comment (jgallen23) [reply index.js:59]: any idea why they are doing that?

Comment (jgallen23) [reply index.js:59]: ping

File: index.js [hunk 3]
```diff
@@ -63,15 +81,14 @@ const response = function(fn, options = {}) {
     }
     const finish = new Date().getTime();
     const duration = finish - start;
-    const query = req.queryStringParameters || req.query;
     const logObject = {
       statusCode,
       path: req.path,
       method,
       duration,
       userAgent: req.headers['user-agent'] || req.headers['User-Agent'] || '',
```
Comment (jgallen23) [inline index.js]: don't need this fallback anymore

File: index.js [hunk 4]
```diff
@@ -63,15 +81,14 @@ const response = function(fn, options = {}) {
...
       referer: req.headers.referer || req.headers.Referer || '',
```
Comment (jgallen23) [inline index.js]: don't need this fallback anymore

**Review by jgallen23 — CHANGES_REQUESTED** [review]
a few comments

File: index.js (hunk 1 above)
Comment (orthagonal) [reply index.js:59]: I think they just like that style?  There's not really a compelling reason to prefer 'ALPN' over 'alpn' in arc-rapptor, it's just going into a user-provided function anyway.

**Pull request merged**